### DISCUSSION TITLE
sigma_offdiag in multilevel susceptibility simulations

### DISCUSSION THREAD

**Discussion by urbanmur** [NanoComp/meep#2461] [Q&A]
Hi,

I want to simulate lasing from dipoles with arbitrary spatial orientation using MultilevelAtom class. It seems like sigma_offdiag cannot be defined, despite being mentioned in the documentation. Is this correct? Is it expected to be added any time soon?

Thank you and best regards,
Urban

**Comment by smartalecH** [comment]
I think the problem is that meep derives and implements the update equations assuming that the conductivity can commute (which saves a layer of fields caching). This means off-diagonal components aren't supported. Even with the current derivation, this theoretically could work in chunks without PML. But that would require some software engineering effort to refactor things. 

You can read more here: http://ab-initio.mit.edu/meep/pml-meep.pdf

It's a bit odd that the python interface implies this is supported though... might be worth a change.
